### PR TITLE
Add cytoolz as dependency

### LINKED ISSUE #11333: Replace `toolz` dependency with `cytoolz`
This makes me all kind of uncomfortable since cytoolz is a separate pacakge and us using it this way is totally backwards. We should rely on it instead directly or remove our footprint using it. Probably larger than this PR though, so worth opening a ticket to look into it.

_Originally posted by @jezdez in https://github.com/conda/conda/pull/11290#discussion_r826836808_

### PR DESCRIPTION
**Pull request by travishathaway**
### Description
This removes the `toolz` module from our vendored dependencies. Instead, we will rely explicitly on `cytoolz` from now on.

Resolves #11333 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/master/news/TEMPLATE)) for the next release's release notes?
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Review by kenodegard — APPROVED** [review]

**Review by kenodegard — CHANGES_REQUESTED** [review]
This warrants a news snippet.

**Review by kenodegard — APPROVED** [review]

**Comment by kenodegard** [comment]
Just reviewed all of our other repos and found this impacts other repos. `conda._vendor.toolz` is used in both [`constructor`](https://github.com/conda/constructor/blob/7a6cf1d65035c41d5c8f4bb519a5eb8cff8ff9e4/constructor/conda_interface.py#L31-L33) and [`conda-libmamba-solver`](https://github.com/conda-incubator/conda-libmamba-solver/blob/640456f70731518f9d50e7c05b285b0a0e526d0c/conda_libmamba_solver/_libmamba.py#L18).

File: conda/activate.py [hunk 1]
```diff
@@ -13,10 +13,11 @@
 import sys
 from textwrap import dedent
 
+from cytoolz.itertoolz import concatv, drop
```
Comment (jakirkham) [inline conda/activate.py:16]: There is a `tlz` namespace that tries `cytoolz` first and falls back to `toolz`. It also works regardless of which is installed. This is what we do in Dask ( https://github.com/dask/dask/pull/5949#issuecomment-595407707 )

```suggestion
from tlz.itertoolz import concatv, drop
```

Comment (kenodegard) [reply conda/activate.py:16]: hm interesting, and I see that `cytoolz` lists `toolz` as a dependency so it'll always be present

Comment (travishathaway) [reply conda/activate.py]: Thanks for the suggestion. I just updated all of our import statements to use `tlz` instead of `cytoolz` 👍

Comment (jakirkham) [reply conda/activate.py]: Thanks Travis! 🙏

**Review by kenodegard — APPROVED** [review]
🚀

**Review by kenodegard — APPROVED** [review]

**Pull request merged**